### DISCUSSION TITLE
Ability to not set version?

### DISCUSSION THREAD

**Discussion by GoodClover** [microsoft/winget-pkgs#3566] [General]
I am planning to ass [JOSM](josm.openstreetmap.de) to winget.
But I have a question.

Is it possible to not set a version?
When creating a manifest for winget you have to set the `Version: `, and therefore the download should be for that fixed version, I have no issues with this but JOSM has a link that always downloads the latest stable release, which would be nice as this repo wouldn't constantly need updating every fortnight.

Also the versioning is just a number (currently `17013`), and that seems to be a no-no in winget.

Here is a link to my current manifest: [Gist/GoodClover/JOSM.yaml](https://gist.github.com/GoodClover/e69580d7ada4bf6c86e348ad2362231b)

This probably isn't the place to ask but seeing as it is always called "JOSM" and rarely by it's full name should I set the name as that?

**Comment by denelon** [comment]
We're looking at an option to support a special sentinel value for manifests with "version: latest". The idea is to support those packages that don't have explicit versions available behind a vanity URL, or those where only a single version is made available. We've also been looking at how we can detect versions from installers where possible, but it would still be problematic if an earlier version had been deprecated, or is no longer available.

**Reply by GoodClover** [reply]
Just saw [winget-cli/issue#338](https://github.com/microsoft/winget-cli/issues/338)

Whilst it appears JOSM has explicit version downloads, this is only for set current versions of interest and the latest so if a new version is released the link to the old one is deleted.
Which would mean winget could no longer use the link and the manifest would need to be changed.
Considering JOSM gets updated frequently this would be an issue.

What should I do in this case?

**Reply by GoodClover** [reply]
Here is JOSM's windows download folder: https://josm.openstreetmap.de/download/windows/
[`josm-setup.exe`](https://josm.openstreetmap.de/download/windows/josm-setup.exe) is always the latest.
[`josm-setup-17013.exe`](https://josm.openstreetmap.de/download/windows/josm-setup-17013.exe) is also the latest, but when a new version is released this link will no longer work (will be replaced with `josm-setup-17014.exe`)
There is also an [archive folder](https://josm.openstreetmap.de/download/Archiv/windows/) but it only goes up to version `16812` and stops for reasons I do not know.
